gnucash/gnucash-cli.cpp

```diff
@@ -126,11 +144,117 @@ may be specified to describe some saved options.\n"
 
 }
 
+struct cli_struct
+{
+    boost::optional<std::string>& script;
+    bool interactive;
+    bool verbose;
+};
+
+static void
+run_guile_cli (void *data, [[maybe_unused]] int argc, [[maybe_unused]] char **argv)
+{
+    auto args = static_cast<cli_struct*>(data);
+    scm_c_use_module ("system repl repl");
+    scm_c_use_module ("gnucash core-utils");
+    scm_c_use_module ("gnucash engine");
+    scm_c_use_module ("gnucash app-utils");
+    scm_c_use_module ("gnucash report");
+    scm_c_use_module ("ice-9 readline");
+    scm_c_eval_string ("(activate-readline)");
+    if (args->script)
+    {
+        if (args->verbose)
+            std::cout << "Running script from " << *args->script << "...";
+        scm_c_primitive_load (args->script->c_str());
+        if (args->verbose)
+            std::cout << "success!" << std::endl;
+    }
+    if (args->interactive)
+    {
+        if (args->verbose)
+            std::cout << "Starting CLI... " << std::endl;
+        scm_c_eval_string ("(start-repl)");
+    }
+    if (gnc_current_session_exist())
```

Comment:
What happens here if the user has made changes to the data but forgot to interactively run a save command ? Will the changes be silently lost or silently save automatically. Neither is very user friendly.

Reply:
agree - i have it in TODO in the opening PR comment.